### PR TITLE
Giant animals drop mutant meat which has negative effects if relied upon

### PR DESCRIPTION
**Pull request by Davi-DeGanne**
### Summary
```SUMMARY: Balance "Giant animals drop mutant meat, has negative effects if relied upon"```

### Purpose of change
Requires #34695 
Requires #34825 
Overhaul of #34636 

#### From a pseudo-realism perspective
Animals that used to be 1 cm³ but are now 1 m³ expanded in surface area by 10,000x and in volume by 1,000,000x which has dire implications on biology, requiring all new organs, body plans, and metabolic rates in most cases (see [this video](https://www.youtube.com/watch?v=f7KSfjv4Oq0) and [part 2](https://www.youtube.com/watch?v=MUWUHf-rzks) for more info). For instance, spiders' respiratory systems are designed around diffusion via background airflow, rather than pumping air using muscles, and would simply not work quickly enough on a large scale.

Our lore explanation for this is that the blob (or blob-based mutagen made by humans/aliens) has done the necessary mutations to make this possible, but in the process, the meat has grown too quickly, becoming loose and spongiform (and therefore unappetizing and less nutritious), and has additionally become contaminated with toxins that are dangerous if you eat too many.

#### From a balance perspective
Butchering a single giant animal can easily feed a survivor for a long time if they can preserve all the meat. Given the abundance of these creatures, caloric concerns are little more than a chore for most of the game for players that know where to look (which is to say, swamps, spider basements, and ant colonies).

### Describe the solution
All giant animals (spiders, ants, bees, wasps, frogs, naked mole-rats, worms, slugs, rattlesnakes, sewer snakes, albino penguins, cockroaches, centipedes, dragonflies, flies, mosquitoes, dermatiks, and locusts) now drop mutant meat and mutant fat. 
#### Mutant meat has:
- Half as many calories/vitamins as normal meat per unit volume
- A gross texture (-5 enjoyability) that persists even through most crafting, with exceptions being jerky and dehydrated.
- A new vitamin, "Toxins" (more on these later). Your body can safely process the toxins from up to 800 calories of mutant meat per day without any adverse effects.

#### Mutant fat has:
- The same calories/vitamins as normal fat per unit volume (spongiform fat didn't make sense to me)
- No gross texture (same logic as above)
- To compensate, has more toxins (the type of toxins I based these on tends to collect in fatty tissue).
Your body can safely process the toxins from up to 550 calories of mutant fat per day without any adverse effects.

(Note that the toxins from fat and meat are the same, so eating 800 calories of meat _and_ 550 of fat per day, you will experience toxin build-up.)
 
#### Explanation of toxins:
- Toxins are loosely based on man-made compounds, primarily insecticides, that don't decompose well, leading to build-up in animal tissue, especially fatty tissue. Symptoms vary, but lowered immune response and neurotoxicity (i.e. seizures, migraines) are common.
- After eating nothing but mutant meat for 2-3 days straight, you will develop mild effects, which _do not_ appear in the @ screen:
  - Infrequent but painful headaches (~1 every 2 days).
  - Lowered immune response, represented by a decaying health value*. You can still keep your health mod at +10 with a modest regimen of healthy food/drink (10 health units per day, or 3-4 servings of pine needle tea). In any case, won't drop health mod below -10 on its own.
- After eating nothing but mutant meat for 4-5 days straight, you will develop moderate effects, which appear as `Concerning symptoms: Your muscles keep twitching strangely.` in the @ screen:
  - Somewhat frequent muscle spasms (~1 every 2 hours) which can make you fall down, drop your wielded item, lower your strength/dexterity for a turn, or cause a lot of recoil if you're aiming a ranged weapon. Falls and weapon drops are prevented on a successful DEX roll. Can be prevented by being on diazepam.
  - Infrequent "drop seizures" which drop you to the ground and prevent you from getting back up for a turn or two. Can be prevented by being on diazepam.
  - The health degradation becomes worse*, you will no longer be able to realistically keep your health mod above -10, but you can still keep it from going lower by eating/drinking healthily as before. In any case, won't drop health mod below -30 on its own.
  - Headaches become twice as common.
- After eating nothing but mutant meat for a week straight, you will develop severe effects, which appear as `Unnerving symptoms: "Your nervous system is malfunctioning, almost like it's being torn apart from the inside.` in the @ screen:
  - Infrequent full body convulsive seizures (~1 every 3 days) which last 1.5 to 4 minutes and will drop you to the ground and nerf your strength and dexterity to the point that you only have a 1/30 chance to stand up each turn. Even if you _do_ stand up, you will be stunned, unable to do much but stumble/flail about randomly. This can of course be game-ending in a combat situation, but can be prevented by being on diazepam.
  - Headaches, drop seizures, and muscle spasms become twice as common.
  - The health degradation becomes worse*, you will no longer be able to realistically keep your health mod above -30, but you can still keep it from going lower by eating/drinking healthily as before. In any case, won't drop health mod below -50 on its own.

\* -- Keep in mind Leukocyte Breeder CBM unconditionally locks health mod to 100, so these effects are moot for characters that have it installed. 

### Describe alternatives you've considered
A lot, see #34636 for starters, this was the product of discussion about that PR on discord, #dev-general on October 12th. A few things I didn't do but kind of want to:
- Expand the mutant meat to non-giant mutants (experimental mutant, evolved mutant, gozu)
- Expand the mutant meat to nether creatures that currently drop normal meat (Gracken, flaming eye, yugg, etc) [we decided to make this "alien meat" and make it really bad, somewhere between mutant meat and tainted meat, in separate PR]
- Give spider eggs bad texture/taste that sticks around through powdering/cooking, or toxins, or both.

### Testing
I have tested the following things and they seem to work as intended:
- `BAD_TASTE` carries correctly to crafted items and the morale malus is correctly included in the description of items.
- Toxin absorption works correctly (besides the fact that vitamin absorption is only ~80% because of a preexisting bug I am working on fixing in a separate PR).
- Drop seizures and full-body seizures work as intended.
- Muscle spasms correctly:
  - Reduce STR/DEX for a turn and max out ranged recoil upon arm spasm.
  - Cause a fall upon leg spasms.
  - Cause wielded item to drop upon arm spasm (and nothing happens if nothing is wielded).
  - Above two effects are avoided more regularly with increasing dexterity.
- All seizure effects are avoided completely by using diazepam.
- `BAD_TASTE` correctly does _not_ carry over to dehydrated meat and jerky.

### Additional context
Here's a table I made that shows the effects of health mod at important breakpoints:

Health mod | Sleep healing | First aid healing | Cold/flu
------------ | ------------- | ------------- | -------------
+10 | 5% faster | 5% faster | ~15% less common
-10 | 5% slower | 2.5% slower | ~15% more common
-30 | 15% slower | 7.5% slower | ~50% more common
-50 | 25% slower | 12.5% slower | 100% more common

### CODE REVIEW

File: src/player_hardcoded_effects.cpp [hunk 1]
```diff
@@ -1301,5 +1304,76 @@ void player::hardcoded_effects( effect &it )
...
+                    add_effect( effect_motor_seizure, td );
+                    add_effect( effect_downed, td );
+                    add_effect( effect_stunned, td );
+                    if( one_in( 3 ) ) {
+                        add_msg_if_player( m_bad, _( "You lose conciousness!" ) );
+                        fall_asleep( td );
+                    }
+                }
+                mod *= 2;
+                [[fallthrough]];
```
Comment (anothersimulacrum) [inline src/player_hardcoded_effects.cpp]: This appears to have been introduced in C++17. This project uses C++14

File: src/player_hardcoded_effects.cpp [hunk 2]
```diff
@@ -1301,5 +1304,76 @@ void player::hardcoded_effects( effect &it )
...
+            case 2:
+                // Myoclonic seizure (muscle spasm)
+                if( one_turn_in( 2_hours / mod ) && !has_effect( effect_valium ) ) {
+                    std::string limb = random_entry<std::vector<std::string>>( { "arm", "hand", "leg" } );
```
Comment (anothersimulacrum) [inline src/player_hardcoded_effects.cpp]: You should use random_body_part instead of this.

File: src/player_hardcoded_effects.cpp [hunk 3]
```diff
@@ -1301,5 +1304,76 @@ void player::hardcoded_effects( effect &it )
...
+                        recoil = MAX_RECOIL;
+                    } else if( limb == "hand" ) {
+                        if( is_armed() && can_unwield( weapon ).success() ) {
+                            if( dice( 4, 4 ) > get_dex() ) {
+                                put_into_vehicle_or_drop( *this, item_drop_reason::tumbling, { remove_weapon() } );
+                            } else {
+                                add_msg_if_player( m_neutral, _( "However, you manage to keep hold of your weapon." ) );
+                            }
+                        }
+                    } else if( limb == "leg" ) {
```
Comment (anothersimulacrum) [inline src/player_hardcoded_effects.cpp]: This will all need to be adjusted as well.

File: data/json/items/comestibles/carnivore.json [hunk 4]
```diff
@@ -109,6 +109,31 @@
     "delete": { "flags": [ "SMOKABLE" ] },
     "vitamins": [ [ "vitA", 0 ], [ "vitC", 0 ], [ "calcium", 0 ], [ "iron", 0 ], [ "vitB", 0 ] ]
   },
+  {
+    "id": "mutant_meat",
+    "type": "COMESTIBLE",
+    "copy-from": "meat",
+    "name": "chunk of mutant meat",
+    "name_plural": "chunks of mutant meat",
+    "description": "Meat from a heavily mutated animal.  It has an unsettling loose and spongy texture, but smells… mostly normal.  Seems safe to eat, if you cook it.",
```
Comment (I-am-Erk) [inline data/json/items/comestibles/carnivore.json]: Suggested alterations for better lore information:
"Meat from a heavily mutated animal.  It has an unsettling loose and spongy texture, but smells… mostly normal.  There are strange tangles and formations in it that don't appear natural at all: bits of bone and hair crusted up inside the muscle, as if trying to form another organism. Still, seems safe to eat, if you cook it and remove the worst parts.",

File: data/json/requirements/cooking_components.json [hunk 5]
```diff
@@ -101,8 +101,14 @@
...
-    "//": "Anything you might consider raw 'red' meat.  About 300-400kcal of meat.",
-    "components": [ [ [ "meat", 1 ], [ "meat_scrap", 10 ], [ "rehydrated_meat", 1 ], [ "meat_canned", 1 ] ] ]
+    "//": "Anything you might consider raw 'red' meat.  About 250mL of meat.",
+    "components": [ [ [ "meat", 1 ], [ "mutant_meat", 1 ], [ "meat_scrap", 10 ], [ "mutant_meat_scrap", 10 ], [ "rehydrated_meat", 1 ], [ "meat_canned", 1 ] ] ]
+  },
+  {
+    "id": "meat_red_raw",
+    "type": "requirement",
+    "//": "Anything you might consider raw, unpreserved 'red' meat.  About 250mL of meat.",
+    "components": [ [ [ "meat", 1 ], [ "mutant_meat", 1 ], [ "meat_scrap", 10 ], [ "mutant_meat_scrap", 10 ] ] ]
```
Comment (anothersimulacrum) [inline data/json/requirements/cooking_components.json]: Linted
```suggestion
    "components": [
      [
        [ "meat", 1 ],
        [ "mutant_meat", 1 ],
        [ "meat_scrap", 10 ],
        [ "mutant_meat_scrap", 10 ]
      ]
    ]
```

File: data/json/items/comestibles/carnivore.json (hunk 4 above)
Comment (DaviBones) [reply data/json/items/comestibles/carnivore.json]: Ooh, flavorful! I like it.

Random thought: It would be really cool if we could gate more detailed descriptions like this behind high intelligence/skill.

**Comment by mlangsdorf** [comment]
If you're up to the challenge and @KorGgenT agrees, the monsters from Magiclysm like owlbears should also be dropping mutant products.

**Comment by KorGgenT** [comment]
yeah, owlbears would probably fit the bill. within magiclysm we could even have a "mutant human meat" option that triggers the negative cannibal stuff and also has the negative effects for trolls, goblins and the like.

File: data/json/items/comestibles/carnivore.json (hunk 4 above)
Comment (I-am-Erk) [reply data/json/items/comestibles/carnivore.json]: Yes it would

**Comment by DaviBones** [comment]
> 
> 
> yeah, owlbears would probably fit the bill. within magiclysm we could even have a "mutant human meat" option that triggers the negative cannibal stuff and also has the negative effects for trolls, goblins and the like.

I can do this, but I've never touched magiclysm so I'd need lists for the things that should be dropping:
- Mutant meat.
- Mutant human meat (I will probably just put this in the base to be dropped by experimental mutants and evolved mutants).
- Alien meat (or perhaps "exotic meat" or something) which will be added in a future PR and will be worse than mutant meat but not as bad as tainted, and will be dropped by i.e. flaming eyes.

**Comment by KorGgenT** [comment]
sounds good. it's fine if magiclysm is a little behind this pr in execution; when you want to go along with it, ping me here or in discord and we'll get the information you need.

File: src/player_hardcoded_effects.cpp [hunk 6]
```diff
@@ -1301,5 +1304,76 @@ void player::hardcoded_effects( effect &it )
...
+                    }
+                }
+                mod *= 2;
+            /* fallthrough */
+            case 2:
+                // Myoclonic seizure (muscle spasm)
+                if( one_turn_in( 2_hours / mod ) && !has_effect( effect_valium ) ) {
+                    std::string limb = random_entry<std::vector<std::string>>( { "arm", "hand", "leg" } );
+                    add_msg_if_player( m_bad,
+                                       _( string_format( "Your %s suddenly jerks in an unexpected direction!", limb ) ) );
```
Comment (Qrox) [inline src/player_hardcoded_effects.cpp]: `limb` in not translated.

File: src/vitamin.cpp [hunk 7]
```diff
@@ -51,7 +53,7 @@ void vitamin::load_vitamin( JsonObject &jo )
 
     vit.id_ = vitamin_id( jo.get_string( "id" ) );
     jo.read( "name", vit.name_ );
-    vit.deficiency_ = efftype_id( jo.get_string( "deficiency" ) );
+    vit.deficiency_ = efftype_id( jo.get_string( "deficiency", "null" ) );
```
Comment (Qrox) [inline src/vitamin.cpp]:
```suggestion
    vit.deficiency_ = efftype_id::NULL_ID();
    jo.read( "deficiency", vit.deficiency_ );
```

File: data/json/items/comestibles/carnivore.json [hunk 8]
```diff
@@ -109,6 +109,32 @@
     "delete": { "flags": [ "SMOKABLE" ] },
     "vitamins": [ [ "vitA", 0 ], [ "vitC", 0 ], [ "calcium", 0 ], [ "iron", 0 ], [ "vitB", 0 ] ]
   },
+  {
+    "id": "mutant_meat",
+    "type": "COMESTIBLE",
+    "copy-from": "meat",
+    "name": "chunk of mutant meat",
+    "name_plural": "chunks of mutant meat",
+    "description": "Meat from a heavily mutated animal.  It has an unsettling loose and spongy texture, but smells… mostly normal.  There are strange tangles and formations in it that don't appear natural at all: bits of bone and hair crusted up inside the muscle, as if trying to form another organism.  Still, seems safe to eat, if you cook it and remove the worst parts.",
```
Comment (Fris0uman) [inline data/json/items/comestibles/carnivore.json]:
```suggestion
    "description": "Meat from a heavily mutated animal.  It has an unsettling loose and spongy texture, but smells… mostly normal.  There are strange tangles and formations in it that don't appear natural at all: bits of bone and hair crusted up inside the muscle, as if trying to form another organism.  You can probably eat it, if you cook it and remove the worst parts.",
```
    
Could you reword it to not say `safe to eat` since it's not? Same for meat scrap.

File: src/vitamin.cpp (hunk 7 above)
Comment (DaviBones) [reply src/vitamin.cpp]: Why is this better? Should I change the line below it too, then?

File: src/player_hardcoded_effects.cpp (hunk 6 above)
Comment (DaviBones) [reply src/player_hardcoded_effects.cpp]: Thank you for your patience, I understand finally; the string has to be translated before formatting so that other languages can properly reorder the sentence structure.

File: data/json/items/comestibles/carnivore.json (hunk 8 above)
Comment (DaviBones) [reply data/json/items/comestibles/carnivore.json]: Does replacing `safe to eat` with `digestible at least` in both cases sound good? I don't want to scare players off too badly, since it _is_ completely safe in moderation.

Comment (Fris0uman) [reply data/json/items/comestibles/carnivore.json]: Yep

File: src/player_hardcoded_effects.cpp (hunk 6 above)
Comment (Qrox) [reply src/player_hardcoded_effects.cpp]: Well... not exactly, the string has to be translated before formatting because only the unformatted string gets extracted. The string extraction utility does not know what the formatted string would be, so it's impossible for it to extract the formatted string to be translated by the translators.

File: src/vitamin.cpp (hunk 7 above)
Comment (Qrox) [reply src/vitamin.cpp]: It avoids the magic string `"null"` and uses `string_id`'s own json deserialization function, which is better for code maintenance (imagine if someone wants to change the way `efftype_id` is read. If it's read using its own deserialization function, only the deserialization function needs to be updated. If it's read by reading a string and using the string to construct `efftype_id`, every piece of code that reads an `efftype_id` has to be changed)

And yes, please also change the line below.

Comment (DaviBones) [reply src/vitamin.cpp]: Makes sense, thanks for the detailed explanation! I will try to avoid using `"null"`, I can definitely see why it would cause code maintenance issues.

File: src/player_hardcoded_effects.cpp (hunk 6 above)
Comment (DaviBones) [reply src/player_hardcoded_effects.cpp]: Updated it, is it correct now?

File: src/player_hardcoded_effects.cpp [hunk 9]
```diff
@@ -1336,8 +1336,8 @@ void player::hardcoded_effects( effect &it )
                 // Myoclonic seizure (muscle spasm)
                 if( one_turn_in( 2_hours / mod ) && !has_effect( effect_valium ) ) {
                     std::string limb = random_entry<std::vector<std::string>>( { "arm", "hand", "leg" } );
```
Comment (Qrox) [inline src/player_hardcoded_effects.cpp]: You also need to add `translate_marker()` around these strings to extract them for translation.

**Comment by dissociativity** [comment]
Should this not also apply to triffid and fungal plant matter which is also a huge source of nutrition?

**Comment by I-am-Erk** [comment]
Not really part of this pr. Anyway I'm pretty sure fungaloids haven't been edible for some time. Less sure about triffids.

**Comment by dissociativity** [comment]
That's fair, just people in the discord were saying it's a little silly to have this apply to giant animals and not inter-dimensional plant-based horrors and how easy it is to harvest many healthy meals this way limitlessly, this PR coupled with more evolved mutant animals which I may work on because hey, that's a pretty cool justification for including them since they'll no longer be infinite free food for players c:

**Comment by RDru** [comment]
If we are at it, also Thorny shambler gives a lot of edible veggies.

**Comment by DaviBones** [comment]
> 
> 
> That's fair, just people in the discord were saying it's a little silly to have this apply to giant animals and not inter-dimensional plant-based horrors and how easy it is to harvest many healthy meals this way limitlessly, this PR coupled with more evolved mutant animals which I may work on because hey, that's a pretty cool justification for including them since they'll no longer be infinite free food for players c:

Nobody said that similarly nerfing plant-based monsters' comestible drops was off the table; I may even take a look at it myself. It's just that, as I-am-erk alluded to, this PR is already quite large as it is; something like that really needs to be done separately.

File: src/player_hardcoded_effects.cpp (hunk 9 above)
Comment (DaviBones) [reply src/player_hardcoded_effects.cpp]: Ok added it, is it all good now?
